semgrep/semgrep/spacegrep.py

```diff
@@ -38,15 +39,32 @@ def run_spacegrep(patterns: List[Pattern], targets: List[Path]) -> dict:
                 str(target),
                 pattern_str,
             ]
-            p = sub_run(cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
-            raw_output = p.stdout
-            raw_error = p.stderr
+            try:
+                p = sub_run(cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+                p.check_returncode()
```

Comment:
Is there a reason to use check_returncode then catch the exception instead of checking non-zero exit code in a conditional?

Reply:
Primarily because I don't know what the return code will be (except for 0), so I'm not sure how to handle any non-zero return codes. My thought was to use this `check_returncode` shorthand, which raises a `CalledProcessError`, and then propagate the error upward with an `except` clause for now.